sound/soc/sof/intel/hda.c

```diff
@@ -71,9 +71,15 @@ int hda_ctrl_dai_widget_setup(struct snd_soc_dapm_widget *w)
 		return ret;
 	}
 
+	/* set flag and clear it after IPC */
+	config->flags |= SOF_DAI_CONFIG_FLAGS_HW_PARAMS;
+
 	/* send DAI_CONFIG IPC */
 	ret = sof_ipc_tx_message(sdev->ipc, config->hdr.cmd, config, config->hdr.size,
 				  &reply, sizeof(reply));
+
+	config->flags &= ~SOF_DAI_CONFIG_FLAGS_HW_PARAMS;
```

Comment:
if we use my proposal from the previous patch, we won't have this bit-clearing instruction but instead before each sending of the configuration we'd be setting one of the three values